nixos/modules/services/misc/asusd.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.asusd;
+  tomlFormat = pkgs.formats.toml {};
+in
+{
+  options = {
+    services.asusd = {
+      enable = mkEnableOption ''
+        Provides an interface for rootless control of some system functions most users wish to control such as fan speeds, keyboard LEDs, graphics modes.
+      '';
+      ledmodes = mkOption {
+        type = tomlFormat.type;
+        description = ''
+          <filename>asusd-ledmodes.toml</filename> as a Nix attribute set.
+          See <link xlink:href="https://gitlab.com/asus-linux/asusctl/-/blob/main/data/asusd-ledmodes.toml">asusd-ledmodes.toml</link> for examples.
+        '';
+        default = {};
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Set the ledmodes to the packaged ledmodes by default.
+    services.asusd.ledmodes = mkDefault (
+      let
+        # Convert packaged asusd-ledmodes.toml to JSON.
+        json = pkgs.runCommand "asusd-ledmodes.json" {
+            nativeBuildInputs = [ pkgs.remarshal ];
+          } ''
+            toml2json "${pkgs.asusctl}/etc/asusd/asusd-ledmodes.toml" "$out"
+          '';
+        # Convert JSON to Nix attribute-set.
+        attrs = builtins.fromJSON (builtins.readFile json);
+      in attrs
+    );
+
+    environment.systemPackages = with pkgs; [ asusctl ];
+    services.dbus.packages = with pkgs; [ asusctl ];
+
+    # Avoiding adding udev rules file of asusctl as-is due to the reference to systemd.
```

Comment:
you can patch the file with substituteInPlace

in the derivation do something like 

```nix
{
  systemd = systemd;

  postPatch = ''
    substituteInPlace Makefile \
        --replace "/usr/bin/systemctl "$systemd/bin/systemctl"